x-pack/filebeat/input/cometd/input_test.go

```diff
@@ -0,0 +1,279 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package cometd
+
+import (
+	"context"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	bay "github.com/elastic/bayeux"
+	finput "github.com/elastic/beats/v7/filebeat/input"
+	"github.com/elastic/beats/v7/filebeat/input/inputtest"
+	"github.com/elastic/beats/v7/libbeat/beat"
+	"github.com/elastic/beats/v7/libbeat/common"
+	"github.com/elastic/beats/v7/libbeat/common/atomic"
+	"github.com/elastic/beats/v7/libbeat/logp"
+)
+
+const (
+	channelName = "first-channel"
+)
+
+var (
+	serverURL string
+)
+
+func TestNewInputDone(t *testing.T) {
+	config := common.MapStr{
+		"channel_name":              channelName,
+		"auth.oauth2.client.id":     "DEMOCLIENTID",
+		"auth.oauth2.client.secret": "DEMOCLIENTSECRET",
+		"auth.oauth2.user":          "salesforce_user",
+		"auth.oauth2.password":      "pwd",
+		"auth.oauth2.token_url":     "https://example.com/token",
+	}
+	inputtest.AssertNotStartedInputCanBeDone(t, NewInput, &config)
+}
+
+func TestMakeEventFailure(t *testing.T) {
+	event := beat.Event{
+		Timestamp: time.Now().UTC(),
+		Fields: common.MapStr{
+			"event": common.MapStr{
+				"id":      "DEMOID",
+				"created": time.Now().UTC(),
+			},
+			"message": "DEMOBODYFAIL",
+		},
+		Private: "DEMOBODYFAIL",
+	}
+	assert.NotEqual(t, event, makeEvent("DEMOID", "DEMOBODY"))
+}
+
+func TestNewInput_Run(t *testing.T) {
+	eventsCh := make(chan beat.Event)
+
+	outlet := &mockedOutleter{
+		onEventHandler: func(event beat.Event) bool {
+			eventsCh <- event
+			return true
+		},
+	}
+	connector := &mockedConnector{
+		outlet: outlet,
+	}
+	var inputContext finput.Context
+
+	var msg bay.TriggerEvent
+	msg.Data.Event.ReplayID = 1234
+	msg.Data.Payload = []byte(`{"CountryIso": "IN"}`)
+	msg.Channel = channelName
+
+	config := map[string]interface{}{
+		"channel_name":              channelName,
+		"auth.oauth2.client.id":     "client.id",
+		"auth.oauth2.client.secret": "client.secret",
+		"auth.oauth2.user":          "user",
+		"auth.oauth2.password":      "password",
+	}
+
+	r := http.HandlerFunc(oauth2Handler)
+	server := httptest.NewServer(r)
+	serverURL = server.URL
+	config["auth.oauth2.token_url"] = server.URL + "/token"
+
+	cfg := common.MustNewConfigFrom(config)
+
+	input, err := NewInput(cfg, connector, inputContext)
+	require.NoError(t, err)
+	require.NotNil(t, input)
+
+	input.Run()
```

Comment:
Should there be a `defer input.Stop()` after each `Run()` call to ensure the input is cleaned up?